hepcrawl/spiders/mit_spider.py

```diff
@@ -0,0 +1,244 @@
+        req = requests.post(url, data=post_data)
+
+        listing_file = get_temporary_file(prefix="MIT_", suffix=".html")
+
+        with open(listing_file, "w") as outfile:
+            outfile.write(req.text)
+
+        call_uri = u"file://{0}".format(listing_file)
+        return call_uri
+
+    @staticmethod
+    def get_authors(node):
+        """Return authors dictionary """
+        authors_raw = node.xpath(
+            "//td[contains(text(), 'dc.contributor.author')]/following-sibling::td[1]/text()").extract()
+        affiliation = node.xpath(
+            "//td[contains(text(), 'dc.contributor.department')]/following-sibling::td[1]/text()").extract_first()
+
+        authors = []
+        pattern = re.compile(r"(.*)\,\s\d{4}.?")  # Might contain birthdate
+        for author in authors_raw:
+            try:
+                author = pattern.search(author).group(1)
+            except AttributeError:
+                pass
+            surname, given_names = split_fullname(author)
+            authdict = {
+                'surname': surname,
+                'given_names': given_names,
+            }
+            if affiliation:
+                authdict["affiliations"] = [{"value": affiliation}]
+            authors.append(authdict)
+
+        return authors
+
+    def add_fft_file(self, pdf_files, file_access, file_type):
+        """Create a structured dictionary to add to 'files' item."""
+        file_dicts = []
+        for link in pdf_files:
+            file_dict = {
+                "access": file_access,
+                "description": self.name,
+                "url": urljoin(self.domain, link),
+                "type": file_type,
+            }
+            file_dicts.append(file_dict)
+        return file_dicts
+
+    @staticmethod
+    def get_thesis_info(node):
+        """Create thesis info dictionary."""
+        date = node.xpath(
+            "//td[contains(text(), 'dc.date.issued')]/following-sibling::td[1]/text()").extract_first()
+        university = node.xpath(
+            "//td[contains(text(), 'dc.publisher')]/following-sibling::td[1]/text()").extract()
+
+        thesis = {
+            "date": date,
+            "university": university,
+            "degree_type": "PhD",
+        }
+        return thesis
+
+    @staticmethod
+    def get_thesis_supervisors(node):
+        """Create a structured supervisor dictionary.
+
+        There might be multiple supervisors.
+        """
+        supervisors_raw = node.xpath(
+            "//td[contains(text(), 'dc.contributor.advisor')]/following-sibling::td[1]/text()").extract()
+        supers = []
+        for supervisor in supervisors_raw:
+            if "and" in supervisor:
+                supers.extend(supervisor.split(" and "))
+            else:
+                supers.append(supervisor)
+
+        supervisors = []
+        for supervisor in supers:
+            surname, given_names = split_fullname(supervisor)
+            supervisors.append({
```

Comment:
Maybe just make a `parse_thesis_supervisors` that calls `parse_authors` and then removes the other fields.